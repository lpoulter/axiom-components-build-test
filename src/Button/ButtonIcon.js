import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Icon from '../Icon/Icon';
import './ButtonIcon.css';

export const ButtonIconRef = 'ButtonIcon';

export default class ButtonIcon extends Component {
  static propTypes = {
    /** Color of the Icon */
    color: PropTypes.string,
    /** SKIP */
    isEnd: PropTypes.bool,
    /** SKIP */
    isStart: PropTypes.bool,
    /** Name of the Icon */
    name: PropTypes.string.isRequired,
    /** Size of the Icon */
    size: PropTypes.string,
  };

  static defaultProps = {
    size: '1rem',
  };

  static typeRef = ButtonIconRef;

  render() {
    const { isEnd, isStart, name, size, color, ...rest } = this.props;
    const classes = classnames('ax-button__icon', {
      'ax-button__icon--start': isStart,
      'ax-button__icon--end': isEnd,
    });

    return (
      <span { ...rest } className={ classes }>
        <Icon name={ name } size={ size } textColor={ color }/>
      </span>
    );
  }
}
