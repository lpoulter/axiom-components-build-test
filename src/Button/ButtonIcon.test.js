import React from 'react';
import renderer from 'react-test-renderer';
import ButtonIcon from './ButtonIcon';

const getComponent = (props = {}) =>
  renderer.create(
    <ButtonIcon { ...props } name="twitter" />
  );

describe('ButtonIcon', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with end spacing', () => {
    const component = getComponent({ isEnd: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with start spacing', () => {
    const component = getComponent({ isStart: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
