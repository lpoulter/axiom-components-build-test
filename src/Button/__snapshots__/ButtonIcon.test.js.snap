// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon renders with defaultProps 1`] = `
<span
  className="ax-button__icon"
>
  <svg
    className="ax-icon ax-icon--twitter"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;

exports[`ButtonIcon renders with end spacing 1`] = `
<span
  className="ax-button__icon ax-button__icon--end"
>
  <svg
    className="ax-icon ax-icon--twitter"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;

exports[`ButtonIcon renders with start spacing 1`] = `
<span
  className="ax-button__icon ax-button__icon--start"
>
  <svg
    className="ax-icon ax-icon--twitter"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;
